OpenRA.Game/Map/CellLayer.cs

```diff
@@ -34,15 +35,35 @@ public CellLayer(TileShape shape, Size size)
 			entries = new T[size.Width * size.Height];
 		}
 
-		public void CopyValuesFrom(CellLayer<T> anotherLayer)
+		// provides a default instance for this type of Layer
+		public static void Init(CellLayer<T> defaultLayer)
 		{
-			if (Size != anotherLayer.Size || Shape != anotherLayer.Shape)
+			defaultCellInfoLayer = defaultLayer;
+		}
+
+        public static void Init(Func<int, int, T> creator, int mapWidth, int mapHeight, TileShape tileShape)
```

Comment:
But with a static Init method, it's coupled to everything because it's globally accessible. What happens once another part of the program needs a CellLayer with different default parameters? It's going to break something else it doesn't even know about.